### DISCUSSION TITLE
Test discussion

### DISCUSSION THREAD

**Discussion by zubyoz** [Cyfrin/foundry-full-course-cu#1116] [General]
Hello, Z here!

**Comment by adityav477** [comment]
who is the owner in :
   ```
function testWithdrawWithSingleFunder() public funded {
        //Arrange
        uint256 startingOwnerBalance = fundMe.getOwner().balance; //to check balance of owner(here it is USER that deployed contract)
        uint256 startingFundMeBalance = address(fundMe).balance; //to check the balance in the fundMe contract

        //Act
        vm.prank(fundMe.getOwner());
        fundMe.withdraw();
        // console.log(deployFundMeAddress);
        // console.log(fundMe.getOwner());
        // console.log(address(this));

        //Assert
        uint256 endingOwnerBalance = fundMe.getOwner().balance;
        // uint256 endingFundMeBalance = address(fundMe).balance;
        assertEq(
            endingOwnerBalance,
            (startingOwnerBalance + startingFundMeBalance)
        );
    }`
```
the deployFundMeAddress is the address of deployFundMe.sol 
i am not able to trace the owner/msg.sender 
thakyou for your time
is it  a random account from the anvil chain??

**Reply by iftikharuddin** [reply]
If you are inquiring about the test, the owner for the next transaction will be `fundMe.getOwner()` because you pranked the user before initiating the withdrawal using `vm.prank(fundMe.getOwner());`. 

This action simulates the scenario where the owner, after being pranked, attempts to withdraw funds. The `vm.prank()` function allows for mocking different roles in the testing environment, and in this case, it sets the user as the owner for the subsequent transaction.

**Reply by talhaEth** [reply]
Because you pranked by using this line of code, `vm.prank(fundMe.getOwner());`
 the owner will be fundMe.getOwner().

you can also do like this.
`address public owner = makeAddr("OWNER");`

`vm.startPrank(OWNER)`

 the line is likely assigning the Ethereum address corresponding to the string "OWNER" to the owner variable.
